samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -900,7 +900,7 @@ class SamzaContainer(
 
   def startTask {
     info("Initializing stream tasks.")
-
+    offsetManager.systemAdmins.values.foreach(_.start())
```

Comment:
This is wrong, shouldn't be starting SystemAdmins by looking into the offsetManager field. This should be tied either to the lifecycle of the entity that created this, i.e. the SamzaContainer or the entity that owns it, i.e. the offsetManager.

Admins should also be started and stopped everywhere they're being created/used in the framework and not just for OffsetManager. If that's too much work, I'd actually agree with Yi and not make a generic start/stop API.

Reply:
+1 to @nickpan47 and @prateekm 's suggestions.

Reply:
Thank you all for the review. I have update the code to address this issue.